test/unit/client/creating_resource_with_meta_test.dart

```diff
@@ -0,0 +1,49 @@
+import 'package:json_api/client.dart';
+import 'package:json_api/document.dart';
+import 'package:json_api/routing.dart';
+import 'package:json_api/server.dart';
+import 'package:test/test.dart';
+
+import '../../helper/expect_resources_equal.dart';
+
+void main() async {
+  final host = 'localhost';
+  final port = 80;
+  final base = Uri(scheme: 'http', host: host, port: port);
+  final routing = StandardRouting(base);
+
+  group('Client-generated ID', () {
+    JsonApiClient client;
+    RoutingClient routingClient;
+
+    setUp(() async {
```

Comment:
Please use this one as the example https://github.com/f3ath/json-api-dart/blob/master/test/unit/client/conrner_cases_test.dart

There is no need to start an actual server for this test. You may use the `TestHttpHandler` to simulate the response. Then use the `TestHttpHandler.requestLog` property to validate the request (check that the document actually has the meta property).

I understand it can be confusing. Feel free to reach out if you have questions.

Reply:
OK, I have done with TestHttpHandler. Thanks a lot 👍